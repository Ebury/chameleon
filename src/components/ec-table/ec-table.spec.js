import { mount, createLocalVue } from '@vue/test-utils';
import EcTable from './ec-table.vue';

function mountTable(props, mountOpts) {
  return mount(EcTable, {
    propsData: {
      columns: [
        {
          name: 'lorem',
        },
        {
          name: 'ipsum',
        },
      ],
      data: [
        ['foo', 'bar'],
        ['widgets', 'doodads'],
      ],
      ...props,
    },
    ...mountOpts,
  });
}

function mountTableAsTemplate(template, props, mountOpts) {
  const localVue = createLocalVue();

  const Component = localVue.extend({
    components: { EcTable },
    template,
  });

  return mount(Component, {
    localVue,
    propsData: { ...props },
    ...mountOpts,
  });
}

describe('EcTable', () => {
  it('should not render if no props are supplied', () => {
    const wrapper = mountTable({
      columns: undefined,
      data: undefined,
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should not render if not provided with any data', () => {
    const wrapper = mountTable({
      columns: undefined,
      data: undefined,
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should not render if not provided with an empty array of data', () => {
    const wrapper = mountTable({
      columns: undefined,
      data: [],
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render as expected if provided with data but no columns', () => {
    const wrapper = mountTable({
      columns: undefined,
      data: [
        ['foo', 'bar'],
        ['widgets', 'doodads'],
      ],
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render as expected if provided with data but no columns, and footer switched on', () => {
    const wrapper = mountTable({
      columns: undefined,
      data: [
        ['foo', 'bar'],
        ['widgets', 'doodads'],
      ],
      showFooter: true,
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render as expected if provided with data and columns', () => {
    const wrapper = mountTable();

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should have class ec-table-cell--text-center if type of column is icon', () => {
    const wrapper = mountTableAsTemplate(
      '<ec-table :columns="columns" :data="data"/>',
      {},
      {
        data() {
          return {
            columns: [
              {
                name: 'lorem',
              },
              {
                name: 'ipsum',
                type: 'icon',
              },
            ],
            data: [
              ['foo', 'bar'],
              ['widgets', 'doodads'],
            ],
          };
        },
      },
    );

    expect(wrapper.findByDataTest('ec-table__cell-0').classes()).not.toContain('ec-table__cell--text-center');
    expect(wrapper.findByDataTest('ec-table__cell-1').classes()).toContain('ec-table__cell--text-center');

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render as expected if provided with data and columns, with footer switched on', () => {
    const wrapper = mountTable({ showFooter: true });

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render as expected if provided with empty row and no columns, with footer switched on', () => {
    const wrapper = mountTable({
      columns: [],
      data: [[]],
      showFooter: true,
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render as expected if provided with rows and columns, with footer switched on and the icon of the tooltip', () => {
    const wrapper = mountTable({
      showFooter: true,
      tooltipConfig: {
        content: 'This is the tooltip info',
        classes: ['ec-tooltip--bg-bright'],
        placement: 'bottom',
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render as expected if provided with rows and columns, with footer switched on, the icon of the tooltip and the title given', () => {
    const wrapper = mountTable({
      showFooter: true,
      tooltipConfig: {
        content: 'This is the tooltip info',
        classes: ['ec-tooltip--bg-bright'],
        placement: 'bottom',
      },
      title: 'Title example',
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render slots as expected', () => {
    const wrapper = mountTable({
      columns: [
        {
          name: 'lorem',
        },
        {
          name: 'ipsum',
        },
      ],
      data: [
        ['foo', 'bar'],
        ['widgets', 'doodads'],
      ],
      showFooter: true,
    },
    {
      scopedSlots: {
        col2: '<p>{{ props.content }}</p>',
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });
});
