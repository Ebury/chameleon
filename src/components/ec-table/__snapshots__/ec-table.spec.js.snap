// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EcTable should have class ec-table-cell--center if type of content is icon 1`] = `
<div
  class="ec-table-scroll-container"
>
  <table
    class="ec-table"
  >
    <!---->
     
    <thead
      class="ec-table-head"
    >
      <tr>
        <th
          class="ec-table-head__cell"
          scope="col"
        >
          lorem
        </th>
        <th
          class="ec-table-head__cell"
          scope="col"
        >
          ipsum
        </th>
      </tr>
    </thead>
     
    <tbody>
      <tr>
        <td
          class="ec-table__cell"
          data-test="ec-table__cell-0"
        >
          
              foo
            
        </td>
        <td
          class="ec-table__cell"
          data-test="ec-table__cell-1"
        >
          
              bar
            
        </td>
        <td
          class="ec-table__cell ec-table__cell--center"
          data-test="ec-table__cell-2"
        >
          
              lorem
            
        </td>
      </tr>
      <tr>
        <td
          class="ec-table__cell"
          data-test="ec-table__cell-0"
        >
          
              widgets
            
        </td>
        <td
          class="ec-table__cell"
          data-test="ec-table__cell-1"
        >
          
              doodads
            
        </td>
      </tr>
    </tbody>
     
    <!---->
  </table>
</div>
`;

exports[`EcTable should not render if no props are supplied 1`] = `<!---->`;

exports[`EcTable should not render if not provided with an empty array of data 1`] = `<!---->`;

exports[`EcTable should not render if not provided with any data 1`] = `<!---->`;

exports[`EcTable should render as expected if provided with data and columns 1`] = `
<div
  class="ec-table-scroll-container"
>
  <table
    class="ec-table"
  >
    <!---->
     
    <thead
      class="ec-table-head"
    >
      <tr>
        <th
          class="ec-table-head__cell"
          scope="col"
        >
          lorem
        </th>
        <th
          class="ec-table-head__cell"
          scope="col"
        >
          ipsum
        </th>
      </tr>
    </thead>
     
    <tbody>
      <tr>
        <td
          class="ec-table__cell"
          data-test="ec-table__cell-0"
        >
          
              foo
            
        </td>
        <td
          class="ec-table__cell"
          data-test="ec-table__cell-1"
        >
          
              bar
            
        </td>
      </tr>
      <tr>
        <td
          class="ec-table__cell"
          data-test="ec-table__cell-0"
        >
          
              widgets
            
        </td>
        <td
          class="ec-table__cell"
          data-test="ec-table__cell-1"
        >
          
              doodads
            
        </td>
      </tr>
    </tbody>
     
    <!---->
  </table>
</div>
`;

exports[`EcTable should render as expected if provided with data and columns, with footer switched on 1`] = `
<div
  class="ec-table-scroll-container"
>
  <table
    class="ec-table"
  >
    <!---->
     
    <thead
      class="ec-table-head"
    >
      <tr>
        <th
          class="ec-table-head__cell"
          scope="col"
        >
          lorem
        </th>
        <th
          class="ec-table-head__cell"
          scope="col"
        >
          ipsum
        </th>
      </tr>
    </thead>
     
    <tbody>
      <tr>
        <td
          class="ec-table__cell"
          data-test="ec-table__cell-0"
        >
          
              foo
            
        </td>
        <td
          class="ec-table__cell"
          data-test="ec-table__cell-1"
        >
          
              bar
            
        </td>
      </tr>
      <tr>
        <td
          class="ec-table__cell"
          data-test="ec-table__cell-0"
        >
          
              widgets
            
        </td>
        <td
          class="ec-table__cell"
          data-test="ec-table__cell-1"
        >
          
              doodads
            
        </td>
      </tr>
    </tbody>
     
    <tfoot
      class="ec-table-footer"
    >
      <tr>
        <td
          class="ec-table-footer__cell"
          colspan="2"
        >
          <div
            class="ec-table-footer__contents"
          >
            <div
              class="ec-table-footer__details"
            >
              2 items.
            </div>
             
            <!---->
          </div>
        </td>
      </tr>
    </tfoot>
  </table>
</div>
`;

exports[`EcTable should render as expected if provided with data but no columns 1`] = `
<div
  class="ec-table-scroll-container"
>
  <table
    class="ec-table"
  >
    <!---->
     
    <!---->
     
    <tbody>
      <tr>
        <td
          class="ec-table__cell"
          data-test="ec-table__cell-0"
        >
          
              foo
            
        </td>
        <td
          class="ec-table__cell"
          data-test="ec-table__cell-1"
        >
          
              bar
            
        </td>
      </tr>
      <tr>
        <td
          class="ec-table__cell"
          data-test="ec-table__cell-0"
        >
          
              widgets
            
        </td>
        <td
          class="ec-table__cell"
          data-test="ec-table__cell-1"
        >
          
              doodads
            
        </td>
      </tr>
    </tbody>
     
    <!---->
  </table>
</div>
`;

exports[`EcTable should render as expected if provided with data but no columns, and footer switched on 1`] = `
<div
  class="ec-table-scroll-container"
>
  <table
    class="ec-table"
  >
    <!---->
     
    <!---->
     
    <tbody>
      <tr>
        <td
          class="ec-table__cell"
          data-test="ec-table__cell-0"
        >
          
              foo
            
        </td>
        <td
          class="ec-table__cell"
          data-test="ec-table__cell-1"
        >
          
              bar
            
        </td>
      </tr>
      <tr>
        <td
          class="ec-table__cell"
          data-test="ec-table__cell-0"
        >
          
              widgets
            
        </td>
        <td
          class="ec-table__cell"
          data-test="ec-table__cell-1"
        >
          
              doodads
            
        </td>
      </tr>
    </tbody>
     
    <tfoot
      class="ec-table-footer"
    >
      <tr>
        <td
          class="ec-table-footer__cell"
          colspan="2"
        >
          <div
            class="ec-table-footer__contents"
          >
            <div
              class="ec-table-footer__details"
            >
              2 items.
            </div>
             
            <!---->
          </div>
        </td>
      </tr>
    </tfoot>
  </table>
</div>
`;

exports[`EcTable should render as expected if provided with empty row and no columns, with footer switched on 1`] = `
<div
  class="ec-table-scroll-container"
>
  <table
    class="ec-table"
  >
    <!---->
     
    <!---->
     
    <tbody>
      <tr />
    </tbody>
     
    <tfoot
      class="ec-table-footer"
    >
      <tr>
        <td
          class="ec-table-footer__cell"
        >
          <div
            class="ec-table-footer__contents"
          >
            <div
              class="ec-table-footer__details"
            >
              1 item.
            </div>
             
            <!---->
          </div>
        </td>
      </tr>
    </tfoot>
  </table>
</div>
`;

exports[`EcTable should render as expected if provided with rows and columns, with footer switched on and the icon of the tooltip 1`] = `
<div
  class="ec-table-scroll-container"
>
  <table
    class="ec-table"
  >
    <!---->
     
    <thead
      class="ec-table-head"
    >
      <tr>
        <th
          class="ec-table-head__cell"
          scope="col"
        >
          lorem
        </th>
        <th
          class="ec-table-head__cell"
          scope="col"
        >
          ipsum
        </th>
      </tr>
    </thead>
     
    <tbody>
      <tr>
        <td
          class="ec-table__cell"
          data-test="ec-table__cell-0"
        >
          
              foo
            
        </td>
        <td
          class="ec-table__cell"
          data-test="ec-table__cell-1"
        >
          
              bar
            
        </td>
      </tr>
      <tr>
        <td
          class="ec-table__cell"
          data-test="ec-table__cell-0"
        >
          
              widgets
            
        </td>
        <td
          class="ec-table__cell"
          data-test="ec-table__cell-1"
        >
          
              doodads
            
        </td>
      </tr>
    </tbody>
     
    <tfoot
      class="ec-table-footer"
    >
      <tr>
        <td
          class="ec-table-footer__cell"
          colspan="2"
        >
          <div
            class="ec-table-footer__contents"
          >
            <div
              class="ec-table-footer__details"
            >
              2 items.
            </div>
             
            <div
              class="ec-table-footer__tooltip"
            >
              <svg
                class="ec-table-footer__tooltip-icon ec-icon ec-has-tooltip"
                data-original-title="null"
                height="16"
                width="16"
              >
                <use
                  xlink:href="#ec-simple-info"
                />
              </svg>
            </div>
          </div>
        </td>
      </tr>
    </tfoot>
  </table>
</div>
`;

exports[`EcTable should render as expected if provided with rows and columns, with footer switched on, the icon of the tooltip and the title given 1`] = `
<div
  class="ec-table-scroll-container"
>
  <table
    class="ec-table"
  >
    <caption>
      Title example
    </caption>
     
    <thead
      class="ec-table-head"
    >
      <tr>
        <th
          class="ec-table-head__cell"
          scope="col"
        >
          lorem
        </th>
        <th
          class="ec-table-head__cell"
          scope="col"
        >
          ipsum
        </th>
      </tr>
    </thead>
     
    <tbody>
      <tr>
        <td
          class="ec-table__cell"
          data-test="ec-table__cell-0"
        >
          
              foo
            
        </td>
        <td
          class="ec-table__cell"
          data-test="ec-table__cell-1"
        >
          
              bar
            
        </td>
      </tr>
      <tr>
        <td
          class="ec-table__cell"
          data-test="ec-table__cell-0"
        >
          
              widgets
            
        </td>
        <td
          class="ec-table__cell"
          data-test="ec-table__cell-1"
        >
          
              doodads
            
        </td>
      </tr>
    </tbody>
     
    <tfoot
      class="ec-table-footer"
    >
      <tr>
        <td
          class="ec-table-footer__cell"
          colspan="2"
        >
          <div
            class="ec-table-footer__contents"
          >
            <div
              class="ec-table-footer__details"
            >
              2 items.
            </div>
             
            <div
              class="ec-table-footer__tooltip"
            >
              <svg
                class="ec-table-footer__tooltip-icon ec-icon ec-has-tooltip"
                data-original-title="null"
                height="16"
                width="16"
              >
                <use
                  xlink:href="#ec-simple-info"
                />
              </svg>
            </div>
          </div>
        </td>
      </tr>
    </tfoot>
  </table>
</div>
`;

exports[`EcTable should render slots as expected 1`] = `
<div
  class="ec-table-scroll-container"
>
  <table
    class="ec-table"
  >
    <!---->
     
    <thead
      class="ec-table-head"
    >
      <tr>
        <th
          class="ec-table-head__cell"
          scope="col"
        >
          lorem
        </th>
        <th
          class="ec-table-head__cell"
          scope="col"
        >
          ipsum
        </th>
      </tr>
    </thead>
     
    <tbody>
      <tr>
        <td
          class="ec-table__cell"
          data-test="ec-table__cell-0"
        >
          
              foo
            
        </td>
        <td
          class="ec-table__cell"
          data-test="ec-table__cell-1"
        >
          <p>
            bar
          </p>
        </td>
      </tr>
      <tr>
        <td
          class="ec-table__cell"
          data-test="ec-table__cell-0"
        >
          
              widgets
            
        </td>
        <td
          class="ec-table__cell"
          data-test="ec-table__cell-1"
        >
          <p>
            doodads
          </p>
        </td>
      </tr>
    </tbody>
     
    <tfoot
      class="ec-table-footer"
    >
      <tr>
        <td
          class="ec-table-footer__cell"
          colspan="2"
        >
          <div
            class="ec-table-footer__contents"
          >
            <div
              class="ec-table-footer__details"
            >
              2 items.
            </div>
             
            <!---->
          </div>
        </td>
      </tr>
    </tfoot>
  </table>
</div>
`;
